src/vm/rejit.cpp

```diff
@@ -613,39 +613,44 @@ HRESULT ReJitManager::UpdateActiveILVersions(
     BOOL fEESuspended = FALSE;
     SHash<CodeActivationBatchTraits>::Iterator beginIter = mgrToCodeActivationBatch.Begin();
     SHash<CodeActivationBatchTraits>::Iterator endIter = mgrToCodeActivationBatch.End();
-    for (SHash<CodeActivationBatchTraits>::Iterator iter = beginIter; iter != endIter; iter++)
-    {
-        CodeActivationBatch * pCodeActivationBatch = *iter;
-        CodeVersionManager * pCodeVersionManager = pCodeActivationBatch->m_pCodeVersionManager;
 
-        int cMethodsToActivate = pCodeActivationBatch->m_methodsToActivate.Count();
-        if (cMethodsToActivate == 0)
-        {
-            continue;
-        }
+    {
+        MethodDescBackpatchInfoTracker::ConditionalLockHolder lockHolder;
```

Comment:
If the lock ordering has to remain this way, to limit the lock scope it would be better to move it into `CodeVersionManager::SetActiveILCodeVersions` to just before where the `TableLockHolder` is instantiated.

Reply:
The lock must be outside of any use of EE thread suspension, which takes and holds the ThreadStore lock.

Reply:
Oh is that because of the coop mode requirements?

Reply:
Actually I don't see why coop mode would require that, and I still don't understand why the lock has to be taken before EE thread suspension, could you elaborate?

Reply:
Its more about the issues around allocating on the GC heap. (Not something that happens in the visitation paths, but since you need to use the same crst to protect the add path, you end up with this problem.)